<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(find multi_robots_stage)/stage_config/maps/rink.pgm 0.04" respawn="false" />

  <node pkg="stage_ros_add_pose_and_crash" type="stageros" name="stageros" args="$(find multi_robots_stage)/stage_config/worlds/circle.world" respawn="false">
    <param name="base_watchdog_timeout" value="0.2"/>
  </node>

  <!-- BEGIN ROBOT 0 -->
  <group ns="robot_0">
   <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/move_base_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/dwa_local_planner_params.yaml" command="load" />

      <param name="global_costmap/robot_base_frame" value="robot_0/base_link"/>
      <param name="local_costmap/global_frame" value="robot_0/odom"/>
      <param name="local_costmap/robot_base_frame" value="robot_0/base_link"/>
    </node> 

    <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
      <param name="tf_prefix" value="robot_0" />
      <param name="odom_frame_id" value="robot_0/odom" />
      <param name="base_frame_id" value="robot_0/base_link" />
      <param name="delta_x" value="-16.1" />
      <param name="delta_y" value="-16.1" />
      <param name="delta_yaw" value="0.0" />
      <param name="global_frame_id" value="map" />
    </node>
  </group>
  <!-- END ROBOT 0 -->

  <!-- BEGIN ROBOT 1 -->
  <group ns="robot_1">

   <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/move_base_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/dwa_local_planner_params.yaml" command="load" />

      <param name="global_costmap/robot_base_frame" value="robot_1/base_link"/>
      <param name="local_costmap/global_frame" value="robot_1/odom"/>
      <param name="local_costmap/robot_base_frame" value="robot_1/base_link"/>

    </node> 

    <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
      <param name="tf_prefix" value="robot_1" />
      <param name="odom_frame_id" value="robot_1/odom" />
      <param name="base_frame_id" value="robot_1/base_link" />
      <param name="delta_x" value="-16.1" />
      <param name="delta_y" value="-16.1" />
      <param name="delta_yaw" value="0.0" />
      <param name="global_frame_id" value="map" />
    </node>
  </group>
  <!-- END ROBOT 1 -->

  <!-- BEGIN ROBOT 2 -->
  <group ns="robot_2">

   <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/move_base_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/dwa_local_planner_params.yaml" command="load" />

      <param name="global_costmap/robot_base_frame" value="robot_2/base_link"/>
      <param name="local_costmap/global_frame" value="robot_2/odom"/>
      <param name="local_costmap/robot_base_frame" value="robot_2/base_link"/>

    </node> 

    <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
      <param name="tf_prefix" value="robot_2" />
      <param name="odom_frame_id" value="robot_2/odom" />
      <param name="base_frame_id" value="robot_2/base_link" />
      <param name="delta_x" value="-16.1" />
      <param name="delta_y" value="-16.1" />
      <param name="delta_yaw" value="0.0" />
      <param name="global_frame_id" value="map" />
    </node>
  </group>
  <!-- END ROBOT 2 -->

  <!-- BEGIN ROBOT 3 -->
  <group ns="robot_3">

   <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/move_base_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/dwa_local_planner_params.yaml" command="load" />

      <param name="global_costmap/robot_base_frame" value="robot_3/base_link"/>
      <param name="local_costmap/global_frame" value="robot_3/odom"/>
      <param name="local_costmap/robot_base_frame" value="robot_3/base_link"/>

    </node> 

    <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
      <param name="tf_prefix" value="robot_3" />
      <param name="odom_frame_id" value="robot_3/odom" />
      <param name="base_frame_id" value="robot_3/base_link" />
      <param name="delta_x" value="-16.1" />
      <param name="delta_y" value="-16.1" />
      <param name="delta_yaw" value="0.0" />
      <param name="global_frame_id" value="map" />
    </node>
  </group>
  <!-- END ROBOT 3 -->

  <!-- BEGIN ROBOT 4 -->
  <group ns="robot_4">

   <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/move_base_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/dwa_local_planner_params.yaml" command="load" />

      <param name="global_costmap/robot_base_frame" value="robot_4/base_link"/>
      <param name="local_costmap/global_frame" value="robot_4/odom"/>
      <param name="local_costmap/robot_base_frame" value="robot_4/base_link"/>

    </node> 

    <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
      <param name="tf_prefix" value="robot_4" />
      <param name="odom_frame_id" value="robot_4/odom" />
      <param name="base_frame_id" value="robot_4/base_link" />
      <param name="delta_x" value="-16.1" />
      <param name="delta_y" value="-16.1" />
      <param name="delta_yaw" value="0.0" />
      <param name="global_frame_id" value="map" />
    </node>
  </group>
  <!-- END ROBOT 4 -->

  <!-- BEGIN ROBOT 5 -->
  <group ns="robot_5">

   <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/move_base_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_robots_stage)/move_base_config/dwa_local_planner_params.yaml" command="load" />

      <param name="global_costmap/robot_base_frame" value="robot_5/base_link"/>
      <param name="local_costmap/global_frame" value="robot_5/odom"/>
      <param name="local_costmap/robot_base_frame" value="robot_5/base_link"/>

    </node> 

    <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
      <param name="tf_prefix" value="robot_5" />
      <param name="odom_frame_id" value="robot_5/odom" />
      <param name="base_frame_id" value="robot_5/base_link" />
      <param name="delta_x" value="-16.1" />
      <param name="delta_y" value="-16.1" />
      <param name="delta_yaw" value="0.0" />
      <param name="global_frame_id" value="map" />
    </node>
  </group>
  <!-- END ROBOT 5 -->

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find multi_robots_stage)/multi_robot.rviz" >
     <remap from="/move_base_simple/goal" to="/robot_0/move_base_simple/goal" />
  </node>

</launch>
